pkg/controller/certificaterequest/certificaterequest_controller.go

```diff
@@ -189,6 +191,26 @@ func (r *CertificateRequestReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	// From cert-manager v1.3 onwards, CertificateRequests must be approved before they are signed.
+	if !viper.GetBool("disable-approval-check") {
+		log.Info("Checking whether CR has been approved", "cr", req.NamespacedName)
+		// Explicitly fail if the certificate request has been denied
+		if cmutil.CertificateRequestIsDenied(&certificateRequest) {
+			msg := "certificate request has been denied, not signing"
+			log.Info(msg, "cr", req.NamespacedName)
+			setReadyCondition(cmmeta.ConditionFalse, cmapi.CertificateRequestReasonFailed, msg)
```

Comment:
It's been mostly decided that we are going to use a different Ready Reason than Failed https://github.com/jetstack/cert-manager/pull/3878. You might want to wait for v1.3.1 for `cmapi.CertificateRequestReasonDenied`, but you can use anything you want.

Reply:
We should also move this line out of the `disable-approval-check` flag check. We optimistically always check for Denial, but can choose to ignore approval.